src/components/EditAvatarModal.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import UserAvatar from "./UserAvatar";
+import { useAppDispatch, useAppSelector } from "../redux/hooks";
+import * as types from "../utilities/types";
+import { Alert, Button, Input, Modal } from "antd";
+import { useState } from "react";
+import { CameraOutlined } from "@ant-design/icons";
+import { ActivityContentImageLink } from "@dsnp/sdk/core/activityContent";
+import { core } from "@dsnp/sdk";
+import path from "path";
+import { setTempIconUri } from "../redux/slices/userSlice";
+
+interface EditAvatarModalProps {
+  setNewIcon: (icon: ActivityContentImageLink[] | undefined) => void;
+}
+
+const EditAvatarModal = ({ setNewIcon }: EditAvatarModalProps): JSX.Element => {
+  const dispatch = useAppDispatch();
+
+  const userId: string | undefined = useAppSelector((state) => state.user.id);
+  const profiles: Record<types.HexString, types.User> = useAppSelector(
+    (state) => state.profiles?.profiles || {}
+  );
+  const user: types.User | undefined = userId ? profiles[userId] : undefined;
+
+  const [isModalVisible, setIsModalVisible] = useState<boolean>(false);
+  const [iconUri, setIconUri] = React.useState<string>("");
+  const [urlErrorMsg, setUrlErrorMsg] = React.useState<string | null>(null);
+  const hasValidPostURI = iconUri && String(iconUri).match(/.+\..+\//);
+
+  const handleNewUri = () => {
+    if (hasValidPostURI) {
+      const extension = path.extname(iconUri).replace(".", "");
+      const imageLink = core.activityContent.createImageLink(
+        iconUri,
+        `image/${extension}`,
```

Comment:
This is a little dicey. There's no guarantee the URL will have the correct extension (or especially that the URL will end with it). Also not clear that the image will be one of the [supported types](https://spec.dsnp.org/ActivityContent/Associated/Attachments#supported-image-mime-types). That level of validation will be tough though.

For this PR you should lower case the extension and then check that it is one of the supported types. If it isn't, it's up to you whether you should reject the request or make it "jpeg" and hope for the best.